src/Dfc.CourseDirectory.Web/Controllers/LarsSearchController.cs

```diff
@@ -1,120 +1,111 @@
+                        .OrderBy(f => f.Text).ToArray()),
+                    new LarsSearchFilterModel("Awarding organisation", unfilteredResult.Facets[nameof(Lars.AwardOrgCode)].Select((f, i) =>
+                        new LarsSearchFilterItemModel(
+                            $"{nameof(LarsSearchRequestModel.AwardOrgCodeFilter)}-{i}",
+                            nameof(LarsSearchRequestModel.AwardOrgCodeFilter),
+                            f.Key.ToString(),
+                            f.Key.ToString(),
+                            (int)(f.Value ?? 0),
+                            request.AwardOrgCodeFilter.Contains(f.Key.ToString())))
+                        .OrderBy(f => f.Text).ToArray())
+                },
+                TotalCount = (int)(result.TotalCount ?? 0),
+                PageNumber = request.PageNo,
+                ItemsPerPage = _larsSearchSettings.ItemsPerPage,
+                PageParamName = _larsSearchSettings.PageParamName,
+                Url = Request.GetDisplayUrl()
+            };
 
-                    filter = filters.FirstOrDefault(f => f.Title == "Qualification level");
-                    foreach (SearchFacet sf in resultAll?.Value?.SearchFacets?.NotionalNVQLevelv2)
-                    {
-                        LarsSearchFilterItemModel fim = filter?.Items
-                                                              ?.FirstOrDefault(m => m.Value == sf.Value);
-                        if (fim != null) //&& filter != null)
-                            fim.Count = sfSearch.NotionalNVQLevelv2
-                                                .FirstOrDefault(f => f.Value == sf.Value)
-                                               ?.Count ?? 0;
-                    }
-
-                    model = new LarsSearchResultModel(
-                        requestModel.SearchTerm,
-                        items,
-                        Request.GetDisplayUrl(),
-                        _larsSearchSettings.PageParamName,
-                        requestModel.PageNo,
-                        _larsSearchSettings.ItemsPerPage,
-                        result.Value.ODataCount ?? 0,
-                        filters);
+            return ViewComponent(nameof(ViewComponents.LarsSearchResult.LarsSearchResult), viewModel);
+        }
 
-                } else {
-                    model = new LarsSearchResultModel(result.Error);
-                }
+        //Todo: Move this to an extension method?
+        private static string FormatAwardOrgCodeSearchFilterItemText(string value)
+        {
+            switch (value.ToUpper())
+            {
+                case "E":
+                    return "Entry level";
+                case "X":
+                    return "Unknown or not applicable";
+                case "H":
+                    return "Higher";
+                case "M":
+                    return "Mixed";
+                default:
+                    return $"Level {value}";
             }
```

Comment:
```suggestion
        private static string FormatAwardOrgCodeSearchFilterItemText(string value) => value switch
            {
                "E" => "Entry level",
                "X" => "Unknown or not applicable",
                "H" => "Higher",
                "M" => "Mixed",
                _ => $"Level {value}"
            };
```

Reply:
Moved to `LarsSearchFilterItemModel` and refactored for shared access between `UnregulatedCoursesController` and `LarsSearchController`.